### PR TITLE
Editorial: Add `Iterator` to global object properties

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -29663,7 +29663,7 @@ <h1>EvalError ( . . . )</h1>
       <p>See <emu-xref href="#sec-native-error-types-used-in-this-standard-evalerror"></emu-xref>.</p>
     </emu-clause>
 
-    <emu-clause id="sec-constructor-properties-of-the-global-object-finnalization-registry">
+    <emu-clause id="sec-constructor-properties-of-the-global-object-finalization-registry">
```
Comment (linusg) [inline spec.html:29666]: Not sure if it's worth adding `oldids` for this thb

Comment (michaelficarra) [reply spec.html:29666]: I'd rather just keep the ID the way it is. But if we change it, yes, we should use an oldid.

Comment (ljharb) [reply spec.html:29666]: I doubt anyone's referenced the typoed URL in the last 24 hours - we should definitely change it, and I don't think it matters if there's an oldid or not.

Comment (michaelficarra) [reply spec.html:29666]: This isn't a new clause.

Comment (michaelficarra) [reply spec.html:29666]: @linusg Let's at least do it in a separate PR please.

Comment (ljharb) [reply spec.html:29666]: ah, then yes it definitely needs an old ID

Comment (linusg) [reply spec.html:29666]: -> https://github.com/tc39/ecma262/pull/3458``

**Review by michaelficarra — APPROVED** [review]

File: spec.html [hunk 2]
```diff
@@ -29698,6 +29698,11 @@ <h1>Int32Array ( . . . )</h1>
       <p>See <emu-xref href="#sec-typedarray-constructors"></emu-xref>.</p>
     </emu-clause>
 
+    <emu-clause id="constructor-properties-of-the-global-object-iterator">
```
Comment (jmdyck) [inline spec.html:29701]:
```suggestion
    <emu-clause id="sec-constructor-properties-of-the-global-object-iterator">
```

Comment (michaelficarra) [reply spec.html:29701]: I noticed this, but I'd really rather not be so strict about it. If we really want to be strict about it, we should build it into ecmarkup.

Comment (linusg) [reply spec.html:29701]: Thanks, not sure why I missed that. Fixed :)

**Review by bakkot — APPROVED** [review]

**Pull request merged**